mlflow/tracking/client.py

```diff
@@ -571,10 +574,15 @@ def start_trace(
             if attributes:
                 mlflow_span.set_attributes(attributes)
             trace_manager = InMemoryTraceManager.get_instance()
-            # Update trace tags both in store and in-memory trace manager
-            self._tracking_client.set_trace_tags(request_id, tags or {})
-            trace_info = self._tracking_client.get_trace_info(request_id)
-            trace_manager.update_trace_info(trace_info)
+            if is_in_databricks_model_serving_environment():
+                # Update trace tags for trace in in-memory trace manager
+                with trace_manager.get_trace(request_id) as trace:
+                    trace.info.tags.update(tags or {})
```

Comment:
I still remove the `exclude_immutable_tags` here because inside serving env we won't pass those tags anyways